recipes/libjxl/all/conanfile.py

```diff
@@ -0,0 +1,127 @@
+from conans import ConanFile, CMake, tools
+import os
+import shutil
+import glob
+
+required_conan_version = ">=1.33.0"
+
+
+class LibjxlConan(ConanFile):
+    name = "libjxl"
+    description = "JPEG XL image format reference implementation"
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/libjxl/libjxl"
+    topics = ("image", "jpeg-xl", "jxl", "jpeg")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("brotli/1.0.9")
+        self.requires("highway/0.12.2")
+        self.requires("lcms/2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["JPEGXL_STATIC"] = not self.options.shared
+        self._cmake.definitions["JPEGXL_ENABLE_BENCHMARK"] = False
+        self._cmake.definitions["JPEGXL_ENABLE_EXAMPLES"] = False
+        self._cmake.definitions["JPEGXL_ENABLE_SJPEG"] = False
+        self._cmake.definitions["JPEGXL_ENABLE_OPENEXR"] = False
+        self._cmake.definitions["JPEGXL_ENABLE_SKCMS"] = False
+        self._cmake.definitions["JPEGXL_ENABLE_TCMALLOC"] = False
+        # This part is unreliable.
+        if tools.cross_building(self):
+            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = \
+                str(self.settings.arch)
```

Comment:
To trigger [`CMAKE_CROSSCOMPILING`](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html), you need to set `CMAKE_SYSTEM_NAME`.

Reply:
I don't quite understand what you are suggesting. Here I only need the `CMAKE_SYSTEM_PROCESSOR` variable, which is used in CMake several times. When building to macOS this variable is empty, so I set it manually.

Reply:
Nevermind then.
I thought you were trying to trigger cmake into cross compiling mode.
Why is it unreliable?

Reply:
It may not be the same value when not cross-compiling. Here the value is set from Conan settings.
I can just delete the comment, actually.

Reply:
Oh, you meant to say that the deault `CMAKE_SYSTEM_PROCESSOR`, set by cmake, is not stable.
That is indeed correct.